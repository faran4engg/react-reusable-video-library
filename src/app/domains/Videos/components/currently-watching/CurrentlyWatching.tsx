import { FC, useCallback, useEffect, useState } from 'react';
import VideoCard from '../video-card/VideoCard';
import { CurrentlyWatchingProps } from './types';
import { VideoWatchSkeleton } from 'app/domains/Common/components/loader';
import { sleep } from 'utils/sleep';

const CurrentlyWatching: FC<CurrentlyWatchingProps> = ({
  currentlyWatching,
  id,
}) => {
  const [loading, setLoading] = useState(true);
  const wrapper = document.getElementById('scroll-elem');

  const wait = useCallback(async () => {
    await sleep(500);
    setLoading(false);
  }, []);

  useEffect(() => {
    wait();
  }, [wait]);

  useEffect(() => {
    if (currentlyWatching?.url) {
      wrapper?.scrollIntoView({
        behavior: 'smooth',
        block: 'start',
      });
    }
  }, [wrapper, currentlyWatching]);

  if (loading || !currentlyWatching) {
    return (
      <div className="col-span-2">
        <div className="h-auto overflow-hidden bg-transparent rounded-lg shadow-lg cursor-pointer ">
          <VideoWatchSkeleton />
        </div>
      </div>
    );
  }

  return (
    <div className="col-span-2" id="scroll-elem">
      <div className="overflow-hidden bg-transparent rounded-lg shadow-lg cursor-pointer">
        <VideoCard
          id={id}
          showMoreDetails
          isLightMode={false}
          showWatchBtn={false}
          video={currentlyWatching}
        />
      </div>
    </div>
  );
};

export default CurrentlyWatching;
