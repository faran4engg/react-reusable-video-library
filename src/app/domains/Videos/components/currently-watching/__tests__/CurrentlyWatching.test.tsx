import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import CurrentlyWatching from '../CurrentlyWatching';

const renderer = createRenderer();

describe('<CurrentlyWatching />', () => {
  const currentlyWatchingObj = {
    id: '-MP_h3jk-12hjkh-3k12-jh3',
    title: 'Title',
    url: 'http://yt.com/watch?v=das90ads',
  };
  it('should render and match the snapshot', () => {
    renderer.render(
      <CurrentlyWatching currentlyWatching={currentlyWatchingObj} />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
