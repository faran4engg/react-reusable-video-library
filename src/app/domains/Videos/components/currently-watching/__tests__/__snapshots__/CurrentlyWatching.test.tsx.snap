// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentlyWatching /> should render and match the snapshot 1`] = `
<div
  className="col-span-2"
>
  <div
    className="h-auto overflow-hidden bg-transparent rounded-lg shadow-lg cursor-pointer "
  >
    <VideoWatchSkeleton />
  </div>
</div>
`;
